### PR TITLE
feat(backend): Send BAPI version through headers

### PR DESCRIPTION
**Pull request by jacekradko**
## Description

Sending BAPI calls with `Clerk-API-Version` header

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/backend/src/constants.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const API_VERSION = 'v1';
 export const USER_AGENT = `${PACKAGE_NAME}@${PACKAGE_VERSION}`;
 export const MAX_CACHE_LAST_UPDATED_AT_SECONDS = 5 * 60;
 export const JWKS_CACHE_TTL_MS = 1000 * 60 * 60;
+export const SUPPORTED_BAPI_VERSION = '2024-10-01';
```
Comment (jacekradko) [inline packages/backend/src/constants.ts:7]:
```suggestion
export const SUPPORTED_BAPI_VERSION = '2021-02-05';
```

File: packages/backend/src/mock-server.ts [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+export function validateHeaders(resolver: HttpResponseResolver): HttpResponseResolver<PathParams, DefaultBodyType> {
+  return input => {
+    const { request } = input;
+
+    if (!request.headers.get('Authorization')) {
+      return HttpResponse.json(null, { status: 401 });
+    }
+    if (!request.headers.get('Clerk-API-Version')) {
+      return HttpResponse.json(null, { status: 400 });
+    }
```
Comment (panteliselef) [inline packages/backend/src/mock-server.ts:19]: Do you think a more descriptive json response can help for debugging here ?

Comment (jacekradko) [reply packages/backend/src/mock-server.ts]: @panteliselef Yeah, that's a good idea. Added error details to the responses

File: packages/backend/src/mock-server.ts [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+    // if (!request.headers.get('User-Agent') || request.headers.get('User-Agent') !== '@clerk/backend@0.0.0-test') {
+    //   return HttpResponse.json(null, { status: 400 });
+    // }
```
[the remark this answers is not in the document]
Comment (jacekradko) [reply packages/backend/src/mock-server.ts]: Enabled it now

File: .changeset/angry-cycles-cheer.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/backend': minor
+---
+
+Seding BAPI version through request headers
```
Comment (BRKalow) [inline .changeset/angry-cycles-cheer.md:5]:
```suggestion
Send API version through request headers.
```

File: package.json [hunk 5]
```diff
@@ -73,6 +73,7 @@
     "@types/jest": "^29.3.1",
     "@types/react": "^18.3.3",
     "@types/react-dom": "^18.3.0",
+    "@vitest/coverage-v8": "2.1.4",
```
Comment (BRKalow) [inline package.json:76]: Does this require any further config?

Comment (jacekradko) [reply package.json:76]: Added it to the vitest config in the package:
```
    coverage: {
      provider: 'v8',
    },
```

**Review by panteliselef — APPROVED** [review]

**Pull request merged**